components.go

```diff
@@ -195,3 +198,42 @@ func (m SelectMenu) MarshalJSON() ([]byte, error) {
 		Type:       m.Type(),
 	})
 }
+
+// TextInput represents text input component.
+type TextInput struct {
+	CustomID    string        `json:"custom_id,omitempty"`
+	Label       string        `json:"label"`
+	Style       TextStyleType `json:"style"`
+	Placeholder string        `json:"placeholder,omitempty"`
+	Value       string        `json:"value,omitempty"`
+	Required    bool          `json:"required"`
+	MinLength   int           `json:"min_length"`
+	MaxLength   int           `json:"max_length,omitempty"`
+}
+
+// Type is a method to get the type of a component.
+func (TextInput) Type() ComponentType {
+	return TextInputComponent
+}
+
+// MarshalJSON is a method for marshaling TextInput to a JSON object.
+func (m TextInput) MarshalJSON() ([]byte, error) {
+	type inputText TextInput
+
+	return json.Marshal(struct {
+		inputText
+		Type ComponentType `json:"type"`
+	}{
+		inputText: inputText(m),
+		Type:      m.Type(),
+	})
+}
+
+// TextInputStyleType is style of text in TextInput component.
+type TextInputStyleType uint
+
+// Text styles
+const (
+	TextInputShort     TextStyleType = 1
+	TextInputParagraph TextStyleType = 2
```

Comment:
```suggestion
	TextInputStyleShort     TextStyleType = 1
	TextInputStyleParagraph TextStyleType = 2
```

Reply:
While it's a convention to name it `TextInputStyleShort`, `TextInputShort` looks better.

Reply:
while it may sound better breaking conventions just hurts usability